src/d_delay.c

```diff
@@ -123,6 +125,7 @@ static t_int *sigdelwrite_perform(t_int *w)
 
 static void sigdelwrite_dsp(t_sigdelwrite *x, t_signal **sp)
 {
+    x->x_blocksize = sp[0]->s_n;
     dsp_add(sigdelwrite_perform, 3, sp[0]->s_vec, &x->x_cspace, sp[0]->s_n);
     x->x_sortno = ugen_getsortno();
     sigdelwrite_checkvecsize(x, sp[0]->s_n);
```

Comment:
`sigdelwrite_checkvecsize` will already set `x_vecsize` to the block size.

you would need to make sure that `sigdelwrite_checkvecsize` comes *before* `sigdelwrite_checkvecsize` in `sigdelread_dsp/sigvd_dsp`

Reply:
> basically, the first object in the DSP chain determines the blocksize

additionally, subsequent calls to `sigdelwrite_checkvecsize` should update `x_vecsize` if `vecsize` is *larger*. This makes sure that we resize the delay line according to the *largest* blocksize.